sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -381,6 +381,10 @@ object ColumnPruning extends Rule[LogicalPlan] {
         p
       }
 
+    // Eliminate the child from the Projects with no references to its child
+    case p @ Project(projectList, child) if p.references.intersect(child.outputSet).isEmpty =>
```

Comment:
Yea. I guess it is changed to 1 row? Because in this case, `projectList` should be literals.